components/suggest/src/yelp.rs

```diff
@@ -509,3 +483,227 @@ impl<'a> From<SuggestionBuilder<'a>> for Suggestion {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::{store::tests::TestStore, testing::*, SuggestIngestionConstraints};
+
+    #[test]
+    fn yelp_functions() -> anyhow::Result<()> {
+        before_each();
+
+        let store = TestStore::new(
+            MockRemoteSettingsClient::default()
+                .with_record(SuggestionProvider::Yelp.record("data-4", json!([ramen_yelp()])))
+                .with_record(SuggestionProvider::Yelp.icon(yelp_favicon())),
+        );
+
+        store.ingest(SuggestIngestionConstraints {
+            providers: Some(vec![SuggestionProvider::Yelp]),
+            ..SuggestIngestionConstraints::all_providers()
+        });
+
+        store.read(|dao| {
+            test_find_modifier(dao, "", Modifier::Pre, FindFrom::First, None)?;
+            test_find_modifier(
```

Comment:
Please also test these modifier queries (neither should match):

```
"bestabc"
"bestabc ramen"
```

Also IMO there's no need to test strings with leading/trailing spaces because both `find_modifier` and `find_subject` take a vec slice, not a string, and I think it's safe to assume that they will be called with slices that were created with `split_whitespace().collect::<Vec<_>>()`, which does not create vec items for leading/trailing spaces. e.g., `"   foo    ".split_whitespace().collect::<Vec<_>>()` is just `["foo"]`, not `["", "foo", ""]`.

If you do want to actually test slices with empty strings, that would be fine with me, but you'll need to modify `test_find_modifier`/`test_find_subject`.